src/main.lua

```diff
@@ -720,6 +721,58 @@ addEventingHandler(ActionMap.BuyRecord, utils.hasMatchingTag("Action", ActionMap
 	})
 end)
 
+addEventingHandler("upgradeName", utils.hasMatchingTag("Action", ActionMap.UpgradeName), function(msg)
+	-- get the name
+	local name = string.lower(msg.Tags.Name)
```

Comment:
We see these kinds of transformations so often that I wonder if we'd save some code lines by having a prepend handler that just sets these on the message as like:
- normalizedName
- normalizedFrom
- timestampMs
Could also move the creation of the ioEvent there too.

Reply:
I agree - that's something we can flush out in a separate PR as it'll be useful on every handler